itests/hive-minikdc/src/test/java/org/apache/hadoop/hive/ql/session/TestAddMetastoreDelegationTokenToUGI.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.session;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.TestRemoteHiveMetaStore;
+import org.apache.hadoop.hive.metastore.TestHiveMetaStore;
+import org.apache.hadoop.hive.metastore.conf.MetastoreConf;
+import org.apache.hadoop.hive.metastore.conf.MetastoreConf.ConfVars;
+import org.apache.thrift.transport.TTransportException;
+import org.apache.hadoop.hive.cli.CliSessionState;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hadoop.hive.metastore.utils.SecurityUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hive.minikdc.MiniHiveKdc;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.assertNotNull;
+
+
+import java.util.List;
+
+/**
+ * TestAddMetastoreDelegationTokenToUGI : Simple base test to add metastore delegation token to UserGroupInformation object
+ */
+public class TestAddMetastoreDelegationTokenToUGI extends TestRemoteHiveMetaStore {
```

Comment:
> Can we move this test to itests/hive-minikdc? You seem to use that package a lot in this test and getting the utils will be easier.

What you mean is that TestAddMetastoreDelegationTokenToUGI is located under itest/hive-minikdc/src/test/java/org/apache/hive/minikdc, instead of itest/hive-minikdc/src/test/java/org/apache/hadoop/hive/ql/session? Or should we add the mentioned test case directly to itest/hive-minikdc/src/test/java/org/apache/hive/minikdc/TestRemoteHiveMetaStoreKerberos.java?

Reply:
Yes, adding tests in `itest/hive-minikdc/src/test/java/org/apache/hive/minikdc/TestRemoteHiveMetaStoreKerberos.java` is better.